ArduCopter/mode.cpp

```diff
@@ -631,6 +631,72 @@ void Mode::land_run_horizontal_control()
         }
     }
 
+    // call attitude controller
+    if (auto_yaw.mode() == AUTO_YAW_HOLD) {
+        // roll & pitch from waypoint controller, yaw rate from pilot
+        attitude_control->input_thrust_vector_rate_heading(loiter_nav->get_thrust_vector(), target_yaw_rate);
+    } else {
+        // roll, pitch from waypoint controller, yaw heading from auto_heading()
+        attitude_control->input_thrust_vector_heading(loiter_nav->get_thrust_vector(), auto_yaw.yaw());
+    }
+}
+
+// Go towards a position commanded by prec land state machine
+// Also let user take over the controls if they want to
+void Mode::land_retry_position(const Vector3f &retry_loc)
```

Comment:
this is a lot of duplicate. I think it should be better that we work out some new logic into the rally point to be able to enforce a land on a rally point setted by the precland library

Reply:
might want to rename this to precland_retry_position() but just a suggestion.. not at all a requirement.